src/vm.c

```diff
@@ -892,6 +906,7 @@ V7_PRIVATE val_t s_concat(struct v7 *v7, val_t a, val_t b) {
   }
   memcpy(s, a_ptr, a_len);
   memcpy(s + a_len, b_ptr, b_len);
+  s[a_len + b_len] = '\0';
```

Comment:
sgtm.

w.r.t the conditional, it's already there isn't it ? It'd be just a matter
of moving the statement at the end of the else branch, right?
(anyway it doesn't matter, it would be nano-optimisation)

On Wed, Mar 4, 2015 at 2:34 PM, Sergey Lyubka <notifications@github.com>
wrote:

> In src/vm.c <https://github.com/cesanta/v7/pull/332#discussion_r25774057>:
>
> > @@ -892,6 +906,7 @@ V7_PRIVATE val_t s_concat(struct v7 *v7, val_t a, val_t b) {
> >    }
> >    memcpy(s, a_ptr, a_len);
> >    memcpy(s + a_len, b_ptr, b_len);
> > +  s[a_len + b_len] = '\0';
>
> I'd keep it like that: that way we stress that regardless of the string
> type, it gets 0 terminated. Also, conditional would generate (arguably)
> longer and slower code. I've put a comment though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cesanta/v7/pull/332/files#r25774057>.
>